apps/judicial-system/backend/src/app/modules/case/case.controller.ts

```diff
@@ -288,16 +289,31 @@ export class CaseController {
       }
     }
 
-    const hasCourtIdUpdate =
-      theCase.courtId && update.courtId && theCase.courtId !== update.courtId
-    if (hasCourtIdUpdate) {
-      update.courtCaseNumber = null
-      update.judgeId = null
-      update.registrarId = null
-      update.state = CaseState.SUBMITTED
+    const requiresCourtTransition =
+      theCase.courtId &&
+      update.courtId &&
+      theCase.courtId !== update.courtId &&
+      theCase.state === CaseState.RECEIVED
+
+    const caseUpdate = requiresCourtTransition
+      ? { ...update, ...transitionCase(CaseTransition.MOVE, theCase, user) }
+      : update
+
+    const updatedCase = await this.caseService.update(theCase, caseUpdate, user) // Never returns undefined
+
+    if (requiresCourtTransition) {
+      this.eventService.postEvent(
+        CaseTransition.MOVE,
+        updatedCase ?? theCase,
+        false,
+        {
+          from: theCase.court?.name,
+          to: updatedCase?.court?.name,
+        },
+      )
```

Comment:
Good that you remembered to post a Slack event, I had not thought of that.

You may actually want consider moving the move transition and event to the update method in the case service. I just noticed that that method is already handling the receive transition, so it may make sense to keep them together in one place.

Reply:
Good catch! Will refactor 👌